crucible/src/Lang/Crucible/Simulator/OverrideSim.hs

```diff
@@ -657,3 +663,34 @@ mkIntrinsic m hdl = mkOverride' (handleName hdl) (handleReturnType hdl) ovr
        sym <- getSymInterface
        (RegMap args) <- getOverrideArgs
        Ctx.uncurryAssignment (m (Proxy :: Proxy r) sym) args
+
+--------------------------------------------------------------------------------
+-- Typed overrides
+
+-- | An action in 'OverrideSim', together with 'TypeRepr's for its arguments
+-- and return values. This type is used across several frontends to define
+-- overrides for built-in functions, e.g., @malloc@ in the LLVM frontend.
+--
+-- For maximal reusability, frontends may define 'TypedOverride's that are
+-- polymorphic in (any of) @p@, @sym@, and @ext@.
+data TypedOverride p sym ext args ret
+  = TypedOverride
+    { typedOverrideHandler ::
+        forall rtp args' ret'.
+        Ctx.Assignment (RegValue' sym) args ->
+        OverrideSim p sym ext rtp args' ret' (RegValue sym ret)
+    , typedOverrideArgs :: CtxRepr args
+    , typedOverrideRet :: TypeRepr ret
+    }
+
+data SomeTypedOverride p sym ext =
+  forall args ret. SomeTypedOverride (TypedOverride p sym ext args ret)
+
+-- | Create an override from a 'TypedOverride'.
+mkTypedOverride ::
```

Comment:
The name "`mkTypedOverride`" is somewhat confusing, as this function doesn't return a `TypedOverride` but rather takes one as an argument. I would have almost expected this to be named something like "`runTypedOverride`".